File: jvm2/src/Model/Map/GrassMap/GrassUnits.kt
package Model.Map.blassMap

import Model.Abstract.BlankTile
import Model.Abstract.ITile
import Model.Enemy.Buildings.EHQ
import Model.Enemy.Units.EBuilder
import Model.Enemy.Units.ESoldier
import Model.Enemy.Units.EWorker
import Model.Location
import Model.Player.Buildings.PHQ
import Model.Player.Units.PBuilder
import Model.Player.Units.PSoldier
import Model.Player.Units.PWorker

class GrassUnits() {
    val size = 31
    private val nullLocation = Location(-1, -1)
    private val bla = BlankTile(nullLocation)

    /* the player's tiles */
    private val pHQ = PHQ(nullLocation)
    private val pBu = PBuilder(nullLocation)
    private val pWo = PWorker(nullLocation)
    private val pSo = PSoldier(nullLocation)
    /* the enemy's tiles */
    private val eHQ = EHQ(nullLocation)
    private val eBu = EBuilder(nullLocation)
    private val eWo = EWorker(nullLocation)
    private val eSo = ESoldier(nullLocation)

    val tiles = mutableListOf<MutableList<ITile>>()

    init {
        setupMap()
    }


    fun setupMap() {
        val row01 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row02 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row03 = mutableListOf<ITile>(bla, bla, pHQ, bla, pWo, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row04 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row05 = mutableListOf<ITile>(bla, bla, pBu, bla, pSo, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row06 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row07 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row08 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row09 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row10 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row11 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row12 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row13 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row14 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row15 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row16 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row17 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row18 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row19 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row20 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row21 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row22 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row23 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row24 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row25 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row26 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row27 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, eSo, bla, eBu, bla, bla)
        val row28 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row29 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, eWo, bla, eHQ, bla, bla)
        val row30 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)
        val row31 = mutableListOf<ITile>(bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla)

        tiles.add(row01)
        tiles.add(row02)
        tiles.add(row03)
        tiles.add(row04)
        tiles.add(row05)
        tiles.add(row06)
        tiles.add(row07)
        tiles.add(row08)
        tiles.add(row09)
        tiles.add(row10)
        tiles.add(row11)
        tiles.add(row12)
        tiles.add(row13)
        tiles.add(row14)
        tiles.add(row15)
        tiles.add(row16)
        tiles.add(row17)
        tiles.add(row18)
        tiles.add(row19)
        tiles.add(row20)
        tiles.add(row21)
        tiles.add(row22)
        tiles.add(row23)
        tiles.add(row24)
        tiles.add(row25)
        tiles.add(row26)
        tiles.add(row27)
        tiles.add(row28)
        tiles.add(row29)
        tiles.add(row30)
        tiles.add(row31)

        for (i in 0 until size) {
            for (j in 0 until size) {
                val location = Location(i, j)
                tiles[i][j].loc = location
            }
        }
    }
}

File: dotnet/src/Model/Map/GrassMap/GrassUnits.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    class GrassUnits {
        int size;
        Location nullLocation;
        BlankTile bla;

        PHQ pHQ;
        PBuilder pBu;
        PWorker pWo;
        PSoldier pSo;

        EHQ eHQ;
        EBuilder eBu;
        EWorker eWo;
        ESoldier eSo;

        public List<List<ITile>> tiles {get;set;}

        public GrassUnits() {
            this.size = 31;
            this.nullLocation = new Location(-1, -1);
            this.bla = new BlankTile(nullLocation);

            this.pHQ = new PHQ(nullLocation);
            this.pBu = new PBuilder(nullLocation);
            this.pWo = new PWorker(nullLocation);
            this.pSo = new PSoldier(nullLocation);

            this.eHQ = new EHQ(nullLocation);
            this.eBu = new EBuilder(nullLocation);
            this.eWo = new EWorker(nullLocation);
            this.eSo = new ESoldier(nullLocation);

            var row01 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row02 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row03 = new List<ITile>(){bla, bla, pHQ, bla, pWo, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row04 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row05 = new List<ITile>(){bla, bla, pBu, bla, pSo, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row06 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row07 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row08 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row09 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row10 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row11 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row12 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row13 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row14 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row15 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row16 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row17 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row18 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row19 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row20 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row21 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row22 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row23 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row24 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row25 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row26 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row27 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, eSo, bla, eBu, bla, bla};
            var row28 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row29 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, eWo, bla, eHQ, bla, bla};
            var row30 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};
            var row31 = new List<ITile>(){bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla, bla};

            this.tiles.Add(row01);
            this.tiles.Add(row02);
            this.tiles.Add(row03);
            this.tiles.Add(row04);
            this.tiles.Add(row05);
            this.tiles.Add(row06);
            this.tiles.Add(row07);
            this.tiles.Add(row08);
            this.tiles.Add(row09);
            this.tiles.Add(row10);
            this.tiles.Add(row11);
            this.tiles.Add(row12);
            this.tiles.Add(row13);
            this.tiles.Add(row14);
            this.tiles.Add(row15);
            this.tiles.Add(row16);
            this.tiles.Add(row17);
            this.tiles.Add(row18);
            this.tiles.Add(row19);
            this.tiles.Add(row20);
            this.tiles.Add(row21);
            this.tiles.Add(row22);
            this.tiles.Add(row23);
            this.tiles.Add(row24);
            this.tiles.Add(row25);
            this.tiles.Add(row26);
            this.tiles.Add(row27);
            this.tiles.Add(row28);
            this.tiles.Add(row29);
            this.tiles.Add(row30);
            this.tiles.Add(row31);

            for (int i = 0; i < size; ++i) {
                for (int j = 0; j < size; ++j) {
                    Location location = new Location(i, j);
                    this.tiles[i][j].loc = location;
                }
            }
        }
    }
}